controllers/azuremachine_reconciler.go

```diff
@@ -34,6 +34,8 @@ import (
 	"sigs.k8s.io/cluster-api-provider-azure/cloud/services/virtualmachines"
 )
 
+var _ azure.Service = (*azureMachineService)(nil)
```

Comment:
@shysank can you please open an issue with the thread we had about service composition to track moving the azure.Service interface to a more generic place since it's not specific to cloud/

Reply:
oh, just now saw this comment. did it already [here](https://github.com/kubernetes-sigs/cluster-api-provider-azure/issues/1075) :)